### PR TITLE
:bug: Resolve the sender panic occurring during concurrent calls

### PR DESCRIPTION
**Pull request by yanmxa**
## Related issue(s)
Reference: https://github.com/cloudevents/sdk-go/pull/1094
Fixes: https://github.com/cloudevents/sdk-go/issues/1093

### CODE REVIEW

File: pkg/cloudevents/generic/options/mqtt/agentoptions.go [hunk 1]
```diff
@@ -111,3 +116,16 @@ func (o *mqttAgentOptions) Protocol(ctx context.Context) (options.CloudEventsPro
 func (o *mqttAgentOptions) ErrorChan() <-chan error {
 	return o.errorChan
 }
+
+func appendSubscriptions(options []paho.SubscribeOptions, option paho.SubscribeOptions) []paho.SubscribeOptions {
+	exist := false
+	for _, sub := range options {
+		if sub.Topic == option.Topic {
```
Comment (skeeey) [inline pkg/cloudevents/generic/options/mqtt/agentoptions.go:123]: please add some log

File: pkg/cloudevents/generic/options/mqtt/agentoptions.go [hunk 2]
```diff
@@ -84,17 +84,22 @@ func (o *mqttAgentOptions) WithContext(ctx context.Context, evtCtx cloudevents.E
...
+				Topic: replaceLast(o.Topics.SourceEvents, "+", o.clusterName), QoS: byte(o.SubQoS),
+			},
 		},
 	}
 
+	// receiving status resync events from all sources
 	if len(o.Topics.SourceBroadcast) != 0 {
-		// receiving status resync events from all sources
-		subscribe.Subscriptions[o.Topics.SourceBroadcast] = paho.SubscribeOptions{QoS: byte(o.SubQoS)}
+		subscribe.Subscriptions = appendSubscriptions(subscribe.Subscriptions, paho.SubscribeOptions{
```
Comment (skeeey) [inline pkg/cloudevents/generic/options/mqtt/agentoptions.go:99]: we may use `append` here directly? the `subscribe` should be always a new object, right?

File: pkg/cloudevents/generic/options/mqtt/agentoptions.go (hunk 1 above)
Comment (yanmxa) [reply pkg/cloudevents/generic/options/mqtt/agentoptions.go:123]: append the subscription directly. remove the function

**Comment by skeeey** [comment]
LGTM

**Pull request merged**